src/atoms/ContextMenu.ts

```diff
@@ -0,0 +1,40 @@
+import { useAtom } from "jotai";
+import { atomWithReset, useResetAtom, useUpdateAtom } from "jotai/utils";
+
+export type SelectedCell = {
+  rowIndex: number;
+  colIndex: number;
+};
+
+export type anchorEl = HTMLElement;
+
+const selectedCellAtom = atomWithReset<SelectedCell | null>(null);
+const anchorEleAtom = atomWithReset<HTMLElement | null>(null);
+
+export function useSetAnchorEle() {
+  const setAnchorEle = useUpdateAtom(anchorEleAtom);
+  return { setAnchorEle };
+}
+
+export function useSetSelectedCell() {
+  const setSelectedCell = useUpdateAtom(selectedCellAtom);
+  return { setSelectedCell };
+}
+
+export function useContextMenuAtom() {
+  const [anchorEle] = useAtom(anchorEleAtom);
+  const [selectedCell] = useAtom(selectedCellAtom);
+  const resetAnchorEle = useResetAtom(anchorEleAtom);
+  const resetSelectedCell = useResetAtom(selectedCellAtom);
+
+  const resetContextMenu = async () => {
+    await resetAnchorEle();
+    await resetSelectedCell();
+  };
+
+  return {
+    anchorEle,
+    selectedCell,
+    resetContextMenu,
+  };
+}
```

Comment:
I’m not sure of the benefits of creating custom hooks here, instead of accessing the atoms directly using `useUpdateAtom` in `Table` for example

Reply:
I don't recall where I saw. But I pick up this pattern a while back because it's used to hide the imports 


```jsx
//no custom hook
import { anchorEleAtom } from '@src/store/contextmenu'
import { useAtom } from 'jotai'

const [anchorEle] = useAtom(anchorEleAtom)

```

//custom hooks
```jsx
import { useContextMenu } from '@src/store/contextmenu'

const { anchorEle } = useContextMenu()
```

it's purely syntactic sugar I think. We can remove it if necessary